apps/cbportal/base/CBGrid/.macros/wiki/search/3_search.py

```diff
@@ -0,0 +1,263 @@
+# default page number
+DEFAULT_PAGE_NR = 1
+
+# range of pages shown in page controls
+PAGE_RANGE = 5
+
+
+FILTERS = {
+    "account": {
+        "url": "/CBGrid/account?id=",
+        "search_fields": ["name"],
+        "result_fields": ["id", "name", "status", "creationTime", "acl.userGroupId"],
+        "namespace": "cloudbroker",
+        "label": "Account",
+    },
+    "cloudspace": {
+        "url": "/CBGrid/cloud%20space?id=",
+        "search_fields": ["name", "externalnetworkip"],
+        "result_fields": [
+            "id",
+            "name",
+            "status",
+            "accountId",
+            "creationTime",
+            "networkId",
+            "externalnetworkip",
+            "acl.userGroupId",
+        ],
+        "namespace": "cloudbroker",
+        "label": "Cloud Space",
+    },
+    "vmachine": {
+        "url": "/CBGrid/Virtual%20Machine?id=",
+        "search_fields": ["name", "nics.ipAddress"],
+        "result_fields": [
+            "id",
+            "name",
+            "status",
+            "nics.ipAddress",
+            "stackId",
+            "cloudspaceId",
+        ],
+        "namespace": "cloudbroker",
+        "label": "Virtual Machine",
+    },
+    "image": {
+        "url": "/CBGrid/image?id=",
+        "search_fields": ["name", "type"],
+        "result_fields": ["id", "name", "status", "type"],
+        "namespace": "cloudbroker",
+        "label": "Image",
+    },
+    "disk": {
+        "url": "/CBGrid/disk?id=",
+        "search_fields": ["name", "type"],
+        "result_fields": ["id", "name", "status", "type"],
+        "namespace": "cloudbroker",
+        "label": "Disk",
+    },
+    "externalnetwork": {
+        "url": "/CBGrid/External%20Network?networkid=",
+        "search_fields": ["name", "ips"],
+        "result_fields": ["id", "name", "ips"],
+        "namespace": "cloudbroker",
+        "label": "External Network",
+    },
+    "node": {
+        "url": "/CBGrid/physicalnode?id=",
+        "search_fields": ["name", "ipaddr"],
+        "result_fields": ["id", "name", "status", "ipaddr"],
+        "namespace": "system",
+        "label": "Physical Node",
+    },
+    "user": {
+        "url": "/CBGrid/user?id=",
+        "search_fields": ["id"],
+        "result_fields": ["id", "active", "creationTime", "groups"],
+        "namespace": "system",
+        "label": "User",
+    },
+}
+
+CLICKABLE_FIELDS = {
+    "cloudspaceId": FILTERS["cloudspace"]["url"],
+    "accountId": FILTERS["account"]["url"],
+    "stackId": "/CBGrid/stack?id=",
+    "networkId": FILTERS["externalnetwork"]["url"],
+    "userGroupId": FILTERS["user"]["url"],
+}
+
+
+class Filter(object):
+    def __init__(self, name, dictionary, j):
+        self.name = name
+        self.data = dictionary
+        self._ctx = j.clients.osis.getByInstance("main")
+        self._client = j.clients.osis.getCategory
+
+    def __getattr__(self, attr):
+        return self.data[attr]
+
+    @property
+    def client(self):
```

Comment:
change this to use 
```
from cloudbrokerlib.cloudbroker import db
```
you could do `getattr(getattr(db, self.namespace), self.name)`